compiler/circle-input-names/src/circle-input-names.test.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+
+#include <cstdlib>
+#include <cstring>
+#include <fstream>
+#include <vector>
+#include <regex>
+
+class circle_input_names_test : public ::testing::Test
+{
+protected:
+  // Override Test::SetUp method to run before each test starts
+  void SetUp(void) override;
+
+protected:
+  // Helper functions for tests
+  std::vector<std::string> get_input_names_of(std::string op);
+
+protected:
+  // Dictionary string containing all input names of each op in JSON format
+  std::string _input_names_dict_str;
+};
+
+void circle_input_names_test::SetUp(void)
+{
+  std::string cmd = std::getenv("CIRCLE_INPUT_NAMES_PATH");
+  if (cmd.empty())
+  {
+    throw std::runtime_error("CIRCLE_INPUT_NAMES_PATH is not found");
+  }
+
+  FILE *fp = popen(cmd.c_str(), "r");
```

Comment:
(Optional) Just as a minor comment, I think we can revisit the program not to use popen function in the test by making a function that returns the stringstream. Which makes a test flow simpler.

```c++
int main()
{
  #  ..
  auto ss_input_names = # ..
  std::cout << ss_input_names.str();

  return 0;
}
```

But, this program seems not to have high maintainability. So, it's not that important.

Reply:
If you don't mind, could you elaborate more about it?
Is there a way to redirect forked process standard output to stream buffer?

Reply:
@seanshpark explained about it in offline ;)
Now, I understand what you meant, Thanks 👍